force-app/main/default/classes/OpportunityTriggerHandler.cls

```diff
@@ -0,0 +1,184 @@
+public with sharing class OpportunityTriggerHandler extends TriggerHandler {
+
+    private List<Task> tasksToCreate; // Maintain a list of tasks to insert
+
+    public OpportunityTriggerHandler() {
+        super();
+        this.tasksToCreate = new List<Task>();
+    }
+
+    public override void beforeInsert() {
+        setDefaultOpportunityType((List<Opportunity>) Trigger.new);
+    }
+
+    public override void beforeUpdate() {
+        validateOpportunityAmount((List<Opportunity>) Trigger.new);
+        setPrimaryContact((List<Opportunity>) Trigger.new);
+        appendStageChangeToDescription((List<Opportunity>) Trigger.new, (Map<Id, Opportunity>) Trigger.oldMap);
+    }
+
+    public override void beforeDelete() {
+        preventClosedOpportunityDeletion((List<Opportunity>) Trigger.old);
+        preventClosedWonBankingDeletion((List<Opportunity>) Trigger.old);
+    }
+
+    public override void afterInsert() {
+        createTasksForNewOpportunities((List<Opportunity>) Trigger.new);
+        try {
+            insertTasks(); // Perform DML operation outside of trigger handler
+        } catch (DmlException e) {
+            System.debug('Error inserting tasks: ' + e.getMessage());
+            // Handle exception as needed (e.g., log, retry, notify user)
+        }
+    }
+
+    public override void afterUpdate() {
+        appendStageChangeToDescription((List<Opportunity>) Trigger.new, (Map<Id, Opportunity>) Trigger.oldMap);
+        try {
+            insertTasks(); // Perform DML operation outside of trigger handler
+        } catch (DmlException e) {
+            System.debug('Error inserting tasks: ' + e.getMessage());
+            // Handle exception as needed (e.g., log, retry, notify user)
+        }
+    }
+
+    public override void afterDelete() {
+        notifyOwnersOpportunityDeleted(Trigger.old);
+        // Perform other DML operations if necessary
+    }
+
+    public override void afterUndelete() {
+        assignPrimaryContact((Map<Id, Opportunity>) Trigger.newMap);
+        // Perform other DML operations if necessary
+    }
+
+    private void setDefaultOpportunityType(List<Opportunity> opps) {
+        for (Opportunity opp : opps) {
+            if (opp.Type == null) {
+                opp.Type = 'New Customer';
+            }
+        }
+    }
+
+    private void validateOpportunityAmount(List<Opportunity> opps) {
+        for (Opportunity opp : opps) {
+            if (opp.Amount < 5000) {
+                opp.addError('Opportunity amount must be greater than 5000');
```

Comment:
Try moving these values of 5000 and the string for the error message into a top level variable so it's easier to understand and change in the future